Investigate problems with AMD support when running on macOS

Dear Robert and @drfinkus,

regarding your recent improvement through #5, I believe you might be interested in this outcome. After pulling from master, I get
```
$ python setup.py develop
running develop
running egg_info
creating cpufeature.egg-info
writing cpufeature.egg-info/PKG-INFO
writing dependency_links to cpufeature.egg-info/dependency_links.txt
writing top-level names to cpufeature.egg-info/top_level.txt
writing manifest file 'cpufeature.egg-info/SOURCES.txt'
reading manifest file 'cpufeature.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'README.rst'
writing manifest file 'cpufeature.egg-info/SOURCES.txt'
running build_ext
building 'cpufeature.extension' extension
creating build
creating build/temp.macosx-10.13-x86_64-3.8
creating build/temp.macosx-10.13-x86_64-3.8/cpufeature
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Icpufeature -I/Users/amo/.pyenv/versions/3.8.0/include/python3.8 -c cpufeature/cpu_x86.c -o build/temp.macosx-10.13-x86_64-3.8/cpufeature/cpu_x86.o
In file included from cpufeature/cpu_x86.c:16:
cpufeature/cpu_x86_linux.c:13:10: fatal error: 'sys/sysinfo.h' file not found
#include <sys/sysinfo.h>
         ^~~~~~~~~~~~~~~
1 error generated.
error: command 'clang' failed with exit status 1
```

```
$ clang --version
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

With kind regards,
Andreas.

--- a/cpufeature/cpu_x86_linux.c
+++ b/cpufeature/cpu_x86_linux.c
@@ -10,7 +10,18 @@
 #include "cpu_x86.h"
 #include <pthread.h>
 #include <unistd.h>
+
+#ifdef __linux__
 #include <sys/sysinfo.h>
+#endif
+
+#if defined(__DragonFly__) || \
+    defined(__OpenBSD__)   || \
+    defined(__FreeBSD__)   || \
+    defined(__NetBSD__)    || \
+    defined(__APPLE__)
+#include <sys/sysctl.h>
+#endif
 
 void cpuid(int32_t out[4], int32_t eax, int32_t ecx) {
     // For GCC-specific cpuid

